go/src/lib.rs

```diff
@@ -573,3 +587,67 @@ pub unsafe extern "C" fn command(
         }
     });
 }
+
+fn get_route(route: Routes, cmd: Option<&Cmd>) -> Option<RoutingInfo> {
+    use glide_core::command_request::routes::Value;
+    let Some(route) = route.value.and_then(|route| Some(route)) else {
+        return None;
+    };
+    let get_response_policy = |cmd: Option<&Cmd>| {
+        cmd.and_then(|cmd| {
+            cmd.command()
+                .and_then(|cmd| ResponsePolicy::for_command(&cmd))
+        })
+    };
+    match route {
+        Value::SimpleRoutes(simple_route) => {
+            let simple_route = simple_route.enum_value().unwrap();
+            match simple_route {
+                SimpleRoutes::AllNodes => Some(RoutingInfo::MultiNode((
+                    MultipleNodeRoutingInfo::AllNodes,
+                    get_response_policy(cmd),
+                ))),
+                SimpleRoutes::AllPrimaries => Some(RoutingInfo::MultiNode((
+                    MultipleNodeRoutingInfo::AllMasters,
+                    get_response_policy(cmd),
+                ))),
+                SimpleRoutes::Random => {
+                    Some(RoutingInfo::SingleNode(SingleNodeRoutingInfo::Random))
+                }
+            }
+        }
+        Value::SlotKeyRoute(slot_key_route) => Some(RoutingInfo::SingleNode(
+            SingleNodeRoutingInfo::SpecificNode(Route::new(
+                redis::cluster_topology::get_slot(slot_key_route.slot_key.as_bytes()),
+                get_slot_addr(&slot_key_route.slot_type),
+            )),
+        )),
+        Value::SlotIdRoute(slot_id_route) => Some(RoutingInfo::SingleNode(
+            SingleNodeRoutingInfo::SpecificNode(Route::new(
+                slot_id_route.slot_id as u16,
+                get_slot_addr(&slot_id_route.slot_type),
+            )),
+        )),
+        Value::ByAddressRoute(by_address_route) => match u16::try_from(by_address_route.port) {
+            Ok(port) => Some(RoutingInfo::SingleNode(SingleNodeRoutingInfo::ByAddress {
+                host: by_address_route.host.to_string(),
+                port,
+            })),
+            Err(_) => {
+                // TODO: Handle error propagation.
```

Comment:
todo reminder. Do you want to do it in this PR? If no, please create a task.

Reply:
Created: https://github.com/valkey-io/valkey-glide/issues/2857.